### PR TITLE
Make PathStoreRecord updating cleaner.

### PR DESCRIPTION
**Pull request by kormat**
Also:
- Rename CBS.beacons to CBS.core_beacons, to reduce confusion with
  LBS.beacons, which uses a different data structure.
- Add a short_desc method to PathSegment, to make it easier to correlate
  PCB propagation/errors/etc.
  <a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/398%23issuecomment-144670330%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/398%23discussion_r40904494%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/398%23discussion_r40904679%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/398%23discussion_r40904836%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/398%23issuecomment-144706109%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/398%23discussion_r40920411%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/398%23issuecomment-144670330%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%40syclops%20%40pszalach%20%22%2C%20%22created_at%22%3A%20%222015-10-01T09%3A35%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22lgtm%20%28finally%20pcb%20has%20%60short_desc%60%21%21%3A-%29%20%22%2C%20%22created_at%22%3A%20%222015-10-01T11%3A55%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8159928%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/pszalach%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%200e42fb98b7b87df9c3be6c4f35a6947d3e8d9606%20lib/path_store.py%2024%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/398%23discussion_r40904494%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20also%20needs%20to%20happen%20when%20the%20best%20path%20segments%20are%20selected%20%28i.e.%20in%20%60get_best_segments%60%29.%22%2C%20%22created_at%22%3A%20%222015-10-01T11%3A49%3A40Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1296929%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/syclops%22%7D%7D%2C%20%7B%22body%22%3A%20%22Done.%22%2C%20%22created_at%22%3A%20%222015-10-01T14%3A38%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/path_store.py%3AL227-254%22%7D%2C%20%22Pull%200e42fb98b7b87df9c3be6c4f35a6947d3e8d9606%20lib/path_store.py%2030%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/398%23discussion_r40904679%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Why%20is%20this%20method%20public%3F%20Will%20it%20be%20used%20outside%20%60add_segment%60%3F%20If%20not%2C%20I%20think%20it%20makes%20more%20sense%20to%20make%20it%20private.%22%2C%20%22created_at%22%3A%20%222015-10-01T11%3A52%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1296929%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/syclops%22%7D%7D%2C%20%7B%22body%22%3A%20%22%60add_segment%60%20is%20in%20another%20class%2C%20so%20this%20is%20being%20access%20publicly.%22%2C%20%22created_at%22%3A%20%222015-10-01T11%3A55%3A33Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/path_store.py%3AL227-254%22%7D%7D%7D'></a>
  <a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/398#issuecomment-144670330'>General Comment</a></b>
- <a href='https://github.com/kormat'></a> @syclops @pszalach
- <a href='https://github.com/pszalach'></a> lgtm (finally pcb has `short_desc`!!:-)
- [ ] <a href='#crh-comment-Pull 0e42fb98b7b87df9c3be6c4f35a6947d3e8d9606 lib/path_store.py 24'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/398#discussion_r40904494'>File: lib/path_store.py:L227-254</a></b>
- <a href='https://github.com/syclops'></a> This also needs to happen when the best path segments are selected (i.e. in `get_best_segments`).
- <a href='https://github.com/kormat'></a> Done.
- [ ] <a href='#crh-comment-Pull 0e42fb98b7b87df9c3be6c4f35a6947d3e8d9606 lib/path_store.py 30'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/398#discussion_r40904679'>File: lib/path_store.py:L227-254</a></b>
- <a href='https://github.com/syclops'></a> Why is this method public? Will it be used outside `add_segment`? If not, I think it makes more sense to make it private.
- <a href='https://github.com/kormat'></a> `add_segment` is in another class, so this is being access publicly.

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/398?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/398?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/398'></a>
<a href='#crh-end'></a>

---
This change is [Reviewable](https://reviewable.io/reviews/scionproto/scion/398)

### CODE REVIEW

File: lib/path_store.py [hunk 1]
```diff
@@ -227,20 +227,27 @@ def __init__(self, pcb):
...
         self.id = pcb.get_hops_hash(hex=True)
         self.peer_links = pcb.get_n_peer_links()
         self.hops_length = pcb.get_n_hops()
-        self.delay_time = now - pcb.get_timestamp()
         self.fidelity = 0
         self.disjointness = 0
-        self.last_sent_time = now - self.DEFAULT_OFFSET
-        self.last_seen_time = now
-        self.expiration_time = pcb.get_expiration_time()
+        self.last_sent_time = int(SCIONTime.get_time()) - self.DEFAULT_OFFSET
```
Comment (syclops) [inline lib/path_store.py:235]: This also needs to happen when the best path segments are selected (i.e. in `get_best_segments`).

File: lib/path_store.py [hunk 2]
```diff
@@ -227,20 +227,27 @@ def __init__(self, pcb):
...
         self.guaranteed_bandwidth = 0
         self.available_bandwidth = 0
         self.total_bandwidth = 0
+        self.update(pcb)
+
+    def update(self, pcb):
```
Comment (syclops) [inline lib/path_store.py:241]: Why is this method public? Will it be used outside `add_segment`? If not, I think it makes more sense to make it private.

Comment (kormat) [inline lib/path_store.py:241]: `add_segment` is in another class, so this is being accessed publicly.

File: lib/path_store.py (hunk 1 above)
Comment (kormat) [inline lib/path_store.py:235]: Done.

**Pull request merged**